### PR TITLE
Move to pnpm

### PR DESCRIPTION
**Pull request by samtstern**
This makes the samples significantly smaller and makes our CI run faster. 

#### Size differences

Here are the sizes after running the bootstrap script:

- With `lerna`: 4.5GB
- With `pnpm`:
  - 491MB in `functions-samples`
  - 415MB in `~/.pnpm-store`

#### Speed differences

**Overall**

- Bootstrap phase goes down from 4m to ~15s
- Compile phase goes up from ~1.5m to ~3m

With lerna our CI tests take ~7m, with ~4m in the bootstrap phase ([link](https://github.com/firebase/functions-samples/runs/2655539976?check_suite_focus=true)):

Screen Shot 2021-05-25 at 10 39 55 AM

With pnpm with no concurrency our CI tests take ~4m with ~0.5m in the bootstrap phase ([link](https://github.com/firebase/functions-samples/pull/872/checks?check_run_id=2663840125)):

Screen Shot 2021-05-25 at 10 42 34 AM

With pnpm with concurrency set to 8 (same as Lerna) our CI tests take 3.5m ([link](https://github.com/firebase/functions-samples/pull/872/checks?check_run_id=2663916163)):

Screen Shot 2021-05-25 at 10 49 36 AM

With pnpm with concurrency set to 4 our CI tests take 4.25m ([link](https://github.com/firebase/functions-samples/pull/872/checks?check_run_id=2663969378)):

Screen Shot 2021-05-25 at 10 57 03 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,15 +5,15 @@
...
-    "compile": "lerna run compile --concurrency=8"
+    "bootstrap": "pnpm recursive install",
+    "lint": "pnpm recursive run lint --workspace-concurrency=4",
+    "test": "pnpm recursive run test --workspace-concurrency=4",
+    "compile": "pnpm recursive run compile --workspace-concurrency=4"
   },
   "devDependencies": {
     "eslint": "^6.8.0",
     "eslint-plugin-promise": "^4.2.1",
-    "lerna": "^3.20.2",
```
Comment (jhuleatt) [inline package.json:16]: Does this mean Lerna isn't needed at all any more? If so, want to remove `lerna.json` as well?

File: package.json [hunk 2]
```diff
@@ -5,15 +5,15 @@
   "author": "Firebase (https://firebase.google.com/)",
   "license": "Apache-2.0",
   "scripts": {
-    "bootstrap": "lerna bootstrap --no-ci",
-    "lint": "lerna run lint",
-    "test": "lerna run test",
-    "compile": "lerna run compile --concurrency=8"
+    "bootstrap": "pnpm recursive install",
+    "lint": "pnpm recursive run lint --workspace-concurrency=4",
```
Comment (jhuleatt) [inline package.json:9]: Seems like from your testing, concurrency of 8 was faster than 4 (3.5 minutes vs 4.25 minutes). Is there another reason to use 4?

File: package.json (hunk 1 above)
Comment (samtstern) [reply package.json:16]: Oops yep!

File: package.json (hunk 2 above)
Comment (samtstern) [reply package.json:9]: On my MacBook (and I assume yours) 8-concurrency basically locks up the whole machine and turns it into an airplane :-)

File: pnpm-workspace.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  - quickstarts/uppercase-firestore/functions
+  - quickstarts/uppercase/functions
+  - spotify-auth/functions
+  - stripe/functions
+  - survey-app-update/functions
+  - template-handlebars/functions
+  - text-moderation/functions
+  - url-shortener/functions
+  - username-password-auth/functions
+  - youtube/functions"
```
Comment (jhuleatt) [inline pnpm-workspace.yaml:47]:
```suggestion
  - youtube/functions
```

File: package.json (hunk 2 above)
Comment (jhuleatt) [reply package.json:9]: Haha in that case, 4 sounds good to me!

**Pull request merged**